src/MoBi.Core/Serialization/Services/DeserializedReferenceResolver.cs

```diff
@@ -41,18 +41,30 @@ public void ResolveFormulaAndTemplateReferences(object deserializedObject, IMoBi
          else if (deserializedObject.IsAnImplementationOf<SimulationTransfer>())
          {
             var simulationTransfer = deserializedObject.DowncastTo<SimulationTransfer>();
-            resolveReferences(simulationTransfer.Simulation.DowncastTo<IMoBiSimulation>());
+            var simulation = simulationTransfer.Simulation.DowncastTo<IMoBiSimulation>();
+            updateOutputMappings(simulation, simulationTransfer.OutputMappings);
+            resolveReferences(simulation);
          }
          else if (deserializedObject.IsAnImplementationOf<IMoBiBuildConfiguration>())
          {
-            _builingBlockReferenceUpdater.UpdateTemplatesReferencesIn(deserializedObject.DowncastTo<IMoBiBuildConfiguration>(), project);
+            _buildingBlockReferenceUpdater.UpdateTemplatesReferencesIn(deserializedObject.DowncastTo<IMoBiBuildConfiguration>(), project);
          }
          else if (deserializedObject.IsAnImplementationOf<IModel>())
          {
             resolveReferences(deserializedObject.DowncastTo<IModel>());
          }
       }
 
+      private void updateOutputMappings(IMoBiSimulation simulation, OutputMappings outputMappings)
+      {
+         if (outputMappings == null)
+            return;
+
+         simulation.OutputMappings = outputMappings;
+         //It is required to update the references as the simulation created may have another reference as the one deserialized
+         outputMappings.Each(x => x.UpdateSimulation(simulation));
```

Comment:
I am not quite SURE why this is required. But MoBi does not use the SimulationLoader. Instead something else is used and that creates issue with references